Update .gitignore and solution file with new project configurations

22DPVEmer opened the pull request:
- Expanded .gitignore to include more comprehensive file and directory exclusions
- Added Savanna.Web and Savanna.Infrastructure projects to solution
- Updated project configuration and build settings
- Removed unnecessary file tracking entries
- DB models for the future
1. Images of login and register scaffolded from Identity and landing page for authenticated user:
![image](https://github.com/user-attachments/assets/45dd570a-5b23-4c11-bbb7-f258df07103c)
![image](https://github.com/user-attachments/assets/e213d0d6-3bfb-4cfb-b220-64aa98597887)
![image](https://github.com/user-attachments/assets/90f9ac33-048c-45c3-a3b1-58c9afd81c9d)

2. Database SQL server:
![image](https://github.com/user-attachments/assets/f62cfebb-20df-4f7c-a56d-b6525b4cfe4e)

22DPVEmer committed:
Update .gitignore and solution file with new project configurations

- Expanded .gitignore to include more comprehensive file and directory exclusions
- Added Savanna.Web and Savanna.Infrastructure projects to solution
- Updated project configuration and build settings
- Removed unnecessary file tracking entries

sreeharihm commented on Savanna.Infrastructure/DesignTimeDbContextFactory.cs:
```diff
@@ -0,0 +1,17 @@
+        public ApplicationDbContext CreateDbContext(string[] args)
+        {
+            var optionsBuilder = new DbContextOptionsBuilder<ApplicationDbContext>();
+            optionsBuilder.UseSqlServer("Server=(localdb)\\MSSQLLocalDB;Database=SavannaDB;Trusted_Connection=True;MultipleActiveResultSets=true");
```
move out these connection string from this class and keep it in appsettings.json

22DPVEmer replied:
Moved it out.

sreeharihm commented on Savanna.Web/Program.cs:
```diff
@@ -0,0 +1,64 @@
+    options.UseSqlServer(connectionString));
+
+builder.Services.AddIdentity<ApplicationUser, IdentityRole>(options => {
+        options.SignIn.RequireConfirmedAccount = false;
```
Its better to move all these to configuration

22DPVEmer replied:
Moved the identity to appsettings instead.

sreeharihm replied:
Do it for other options as well

22DPVEmer replied:
There is nothing else to move.

sreeharihm reviewed (commented):
Its better to improve the dashboard section

22DPVEmer committed:
Enhance Dashboard UI and Add Game Statistics

- Redesigned Dashboard page with improved layout and styling
- Added game statistics tracking for saved games
- Implemented extension methods for game save details
- Updated configuration to use appsettings.json for identity options
- Refactored DesignTimeDbContextFactory to use configuration

sreeharihm commented on Savanna.Web/Pages/Dashboard.cshtml.cs:
```diff
@@ -0,0 +1,71 @@
+
+        public static int GetLionCount(this GameSave gameSave)
+        {
+            return gameSave.GameState?.Animals.Count(a => a.AnimalType == "Lion") ?? 0;
```
Remove hard codes and use a constant class

22DPVEmer replied:
Moved it.

sreeharihm reviewed (commented):
Added review comments

22DPVEmer committed:
Centralize error messages and animal type constants

- Replace hardcoded error messages with centralized ErrorMessages constants
- Use AnimalTypes constants for animal type comparisons in Dashboard and Program
- Improve code consistency by referencing centralized constants
